### ISSUE TITLE
Bring back `dotnet migrate` to supported dotnet SDK versions

### ISSUE THREAD

**Issue by dominoFire** [dotnet/sdk#23230]
### Is your feature request related to a problem? Please describe.

I am trying to make try-convert.exe and/or upgrade-assistant support project.json projects but, the actual conversion tool, named `dotnet-migrate` is deprecated and only available up to dotnet 2.x ; see https://docs.microsoft.com/dotnet/core/tools/dotnet-migrate

### Describe the solution you'd like
A clear and concise description of what you want to happen. Include any alternative solutions you've considered.

- Bring back `dotnet-migrate` command to help migrate project.json projects to SDK-style format.
- Call `dotnet-migrate` from try-convert/upgrade-assistant to migrate legacy project.json projects

#### Alternate solutions
- Enable `try-convert.exe` to support using NuGet project.json parsing APIs
- Write custom `try-convert.exe` project.json parsing APIs.

### Additional context

There's a shipped public API in VS to migrate project.json projects to PackageReference in Visual Studio. See https://github.com/NuGet/NuGet.Client/blob/bc11fd124fd87b55532742db2a56f4a324288bcb/src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsProjectJsonToPackageReferenceMigrator.cs#L35

The idea of this issue is to create a CLI experience similar to the VS migration experience.

**Comment by marcpopMSFT** [comment]
Old issue triage: @vijayrkn will the new upgrade assistant work from the CLI? Closing in favor of that and you can link to details about it.

**Comment by baronfel** [comment]
No, it's currently VS IDE only. Usage in VSCode, CLI, and experiences like GitHub Copilot Agent may be considered for the GA release or future versions based on customer feedback, though.
